drivers/i2c/i2c_ll_stm32.c

```diff
@@ -237,6 +237,15 @@ static int i2c_stm32_init(const struct device *dev)
 		return ret;
 	}
 
+#if defined(CONFIG_SOC_STM32F103X8) || defined(CONFIG_SOC_STM32F103XB)
+	/*
+	 * Force i2c reset for STM32 (F101X8/B, F102X8/B), F103X8/B.
+	 * So that they can enter master mode properly.
+	 * Issue described in ES096 2.14.7
+	 */
+	stm32_i2c_reset(dev);
```

Comment:
It would seem to be a little more natural to do this first, before configuring the peripheral, rather than after when everything's already been set up.  For example `stm32_i2c_reset()` may disable transfer interrupts; is there a chance that would be a problem done at this point in the initialization?

Reply:
The transfer interrupt was not enabled until the actual transaction starts. So that should not be a big issue.